errors/errors.go

```diff
@@ -102,38 +105,90 @@ func (e *Error) ErrorV2(additionalInfo interface{}) ErrorV2 {
 }
 
 func GetCode(err error) string {
-	if obj := err.(*Error); obj != nil && obj.Code != " " {
-		return obj.Code
+	var code string
+	defer func() {
+		if r := recover(); r != nil {
+			code = strings.Join(NoneString[:], "")
+		}
+	}()
+	if obj, ok := err.(*Error); ok && obj.Code != "" {
+		code = obj.Code
+	} else if err != nil {
+		code = fmt.Sprintf("%v", err)
 	}
-	return strings.Join(NoneString[:], "")
-}
 
+	return code
+}
 func GetSeverity(err error) Severity {
-	if obj := err.(*Error); obj != nil {
-		return obj.Severity
+	var severity Severity
+	defer func() {
+		if r := recover(); r != nil {
+			severity = None
+			fmt.Println("Recovered from panic:", r)
```

Comment:
I don't know that this is necessary and, at first glance, it seems redundant or not insightful.

Reply:
Thanks for mentioning it was redundant, by mistake I pushed defer on top the assertion logic .
Reason behind using anonymous function is the assertion logic, which I used was giving ```NONE``` on meshkit errors that existed when I released my local meshkit release and used it in go.mod but working fine for local meshkit so new logic is that if we get a panic rather than panic and shutting down the process we catch the panic and returns old value.